pkg/server/server.go

```diff
@@ -776,13 +778,17 @@ func (s *ProxyServer) serveRecvBackend(backend Backend, stream agent.AgentServic
 				if resp.Error != "" {
 					// Dial response with error should not contain a valid ConnID.
 					klog.ErrorS(errors.New(resp.Error), "DIAL_RSP contains failure", "dialID", resp.Random, "agentID", agentID)
+					metrics.Metrics.ObserveDialFailure(metrics.DialFailureErrorResponse)
 					dialErr = true
 				}
 				err := frontend.send(pkt)
 				s.PendingDial.Remove(resp.Random)
 				if err != nil {
 					klog.ErrorS(err, "DIAL_RSP send to frontend stream failure",
 						"dialID", resp.Random, "serverID", s.serverID, "agentID", agentID, "connectionID", resp.ConnectID)
+					if !dialErr { // Avoid double-counting.
+						metrics.Metrics.ObserveDialFailure(metrics.DialFailureSendResponse)
```

Comment:
I don't want to double count. However this is a somewhat interesting case as both the Frontend and Backend connections are both broken. During error handling I would generally expect one of them to be broken, Both seems more interesting.

Reply:
I agree that's an interesting case, but maybe better left to logs? I'm not sure dropping the conditional here would help surface this - or are you suggesting we add a different failure reason to capture this case?